services/src/main/java/io/grpc/services/HealthCheckingLoadBalancerFactory.java

```diff
@@ -171,8 +171,13 @@ public void handleResolvedAddressGroups(
         List<EquivalentAddressGroup> servers, Attributes attributes) {
       Map<String, Object> serviceConfig =
           attributes.get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
-      String serviceName = ServiceConfigUtil.getHealthCheckedServiceName(serviceConfig);
-      helper.setHealthCheckedService(serviceName);
+      try {
+        String serviceName = ServiceConfigUtil.getHealthCheckedServiceName(serviceConfig);
```

Comment:
same comment, I think the best you can do is rethrow as RE, and have it logged top level in LbHelperImpl in ManagedChannelImpl

Reply:
We are not going to have `ManagedChannelImpl` blindly catch and log all RE. It won't be the case either once we have the 2-step parse/apply implementation where the parsing will be moved out of this method. It is wrong for `ManagedChannelImpl` to catch-and-log REs thrown from `handleResolvedAddressGroups()` instead of panicking.

According to @markdroth, C++ just ignores malformed health check config, which is what I am doing in this PR. If you prefer to keep the current behavior, I can rethrow it and let the channel panic. Let me know which one you'd like.